src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_ElectricBlastFurnace.java

```diff
@@ -458,21 +457,27 @@ public int survivalConstruct(ItemStack stackSize, int elementBudget, ISurvivalBu
 
     @Override
     public void onScrewdriverRightClick(byte aSide, EntityPlayer aPlayer, float aX, float aY, float aZ) {
-        isBussesSeparate = !isBussesSeparate;
+        inputSeparation = !inputSeparation;
         GT_Utility.sendChatToPlayer(
                 aPlayer,
-                StatCollector.translateToLocal("GT5U.machines.separatebus") + " " + isBussesSeparate);
+                StatCollector.translateToLocal("GT5U.machines.separatebus") + " " + inputSeparation);
     }
 
     @Override
     public void saveNBTData(NBTTagCompound aNBT) {
         super.saveNBTData(aNBT);
-        aNBT.setBoolean("isBussesSeparate", isBussesSeparate);
     }
 
     @Override
     public void loadNBTData(final NBTTagCompound aNBT) {
         super.loadNBTData(aNBT);
-        isBussesSeparate = aNBT.getBoolean("isBussesSeparate");
+        if (!aNBT.hasKey(INPUT_SEPARATION_NBT_KEY)) {
```

Comment:
I think it should check for isBussesSeparate as the INPUT_SEPARATION_NBT_KEY is always set, so it will never happen
```suggestion
        if (aNBT.hasKey("isBussesSeparate")) {
```

Reply:
No it is not set. If an EBF is loaded for the first time after this is merged it will not have `INPUT_SEPARATION_NBT_KEY` yet. Only after the first loading is that key set and read onwards. With your suggested change existing EBFs would no longer be able to change input separation, because `"isBussesSeparate"` is not deleted from existing NBT, so it will forever load that but never save to it.

Reply:
https://github.com/GTNewHorizons/GT5-Unofficial/blob/master/src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_MultiBlockBase.java#L247
But I dont see any checks here? The key is always set in the NBT Write

Reply:
Also I think that the saveNBTData always starts empty so it should not exist after first save

Reply:
The write doesn't occur before the read. On world load the existing NBT is read, which still contains the old key. When the world is saved the new key will be written to it. Also this is basically just a copy of https://github.com/GTNewHorizons/GT5-Unofficial/blob/fa97edd920c35c0f289bc0d8382d021d21a4a668/src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_ProcessingArray.java#L443 so its proven to work as explained.

Reply:
In short its doing: IF new key doesn't exist that must mean it has the old key. let's use that for this boolean.

This is done after what the super one does so it will never be overridden with the non-existing key for old structures.

Reply:
Well, it doesn't work
![image](https://user-images.githubusercontent.com/53441451/230788429-88b7dff2-ea93-4498-96f2-aa6be2263e75.png)

It will not be executed if updated from stable version (https://github.com/GTNewHorizons/GT5-Unofficial/blob/5.09.41.317/src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_MultiBlockBase.java#L226)
The new key already exist on NBT read.
My suggested solution will work. The nbt saving always starts with empty NBTTagCompound
world load -> loadNBT -> world save -> save new nbt -> the old key does no longer exists

Reply:
Ah I now understand what you mean. It worked for the PA because it was the same update that introduced the flag in the first place.